### PR TITLE
check for changes in assumedly static objects during planning 

### PR DESCRIPTION
**Pull request by tomsilver**
intended for use in pybullet blocks (real robot experiments). this should prevent us from finding plans that have major collisions between the robot and neighboring blocks. 

the `1e-3` tolerance can't be lower because things appear to slightly move around on their own in pybullet.

default to False because some environments (e.g., Cover) don't obey this assumption about skill scope.

### CODE REVIEW

**Review by williamshen-nz — APPROVED** [review]
:shipit: We should probably understand what was going on when we used this flag in the script earlier today

File: predicators/planning.py [hunk 1]
```diff
@@ -499,8 +499,22 @@ def run_low_level_search(task: Task, option_model: _OptionModelBase,
                 discovered_failures[cur_idx - 1] = None
                 num_actions_per_option[cur_idx - 1] = num_actions
                 traj[cur_idx] = next_state
+                # Check if objects that were outside the scope had a change
+                # in state.
+                static_obj_changed = False
+                if CFG.sesame_check_static_object_changes:
+                    static_objs = set(state) - set(nsrt.objects)
+                    for obj in sorted(static_objs):
```
[the remark this answers is not in the document]
Comment (tomsilver) [reply predicators/planning.py:507]: just to make sure that the code runs deterministically. i know that the impact is minimal in this case but can't hurt

**Comment by tomsilver** [comment]
trying to figure out what went wrong on the `blocks-vision` branch with this flag. this seems to work fine:
```
python predicators/main.py --env pybullet_blocks --approach oracle --seed 0 --sesame_check_static_object_changes True --pybullet_robot panda --option_model_use_gui True --option_model_name oracle --option_model_terminate_on_repeat False --blocks_block_size 0.0505
```

**Comment by tomsilver** [comment]
okay the issue in `blocks-vision` was that we were slightly misestimating the z positions of the blocks, which led to them being in collision with the table, and then when physics was applied during `step`, the contact forces would cause all the blocks to move, and this new flag would detect those moves. I'm fixing this for now in `blocks-vision` by making the z position estimation exact by using the known table height.

**Pull request merged**